contracts/Liquidator/Liquidator.sol

```diff
@@ -289,6 +331,39 @@ contract Liquidator is Ownable2StepUpgradeable, ReentrancyGuardUpgradeable, Liqu
         treasuryPercentMantissa = newTreasuryPercentMantissa;
     }
 
+    /**
+     * @notice Sets protocol share reserve contract address
+     * @param protocolShareReserve_ The address of the protocol share reserve contract
+     */
+    function setProtocolShareReserve(address payable protocolShareReserve_) external onlyOwner {
+        _setProtocolShareReserve(protocolShareReserve_);
+    }
+
+    /**
+     * @notice Reduce the reserves of the pending accumulated reserves
+     */
+    function reduceReserves() external {
+        _reduceReservesInternal();
+    }
+
+    function _reduceReservesInternal() internal {
+        uint256 pendingReedemLength_ = pendingRedeem.length;
+        if (pendingReedemLength_ == 0) return;
+        for (uint256 index = pendingReedemLength_ - 1; index >= 0; index--) {
+            address vToken = pendingRedeem[index];
+            uint256 vTokenBalance_ = IVToken(vToken).balanceOf(address(this));
+            if (IVToken(vToken).redeem(vTokenBalance_) != 0) {
```

Comment:
If `redeem` returned value is not 0, then there was an error. I would say, here we have to check `==`, instead of `!=`

Reply:
```
uint256 internal constant NO_ERROR = 0;
...
if (IVToken(vToken).redeem(vTokenBalance_) == NO_ERROR) ...
```

Reply:
Correct I missed equality check.

Reply:
We must also take into account the possibility that the liquidation transaction will be reverted if the market action `redeem` is paused.